src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,24 +1611,1318 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
     ObjectGuid _playerGUID;
 };
 
-void AddSC_silverpine_forest()
+enum BondoftheValkyr
 {
-    /* Vehicles */
+    SPELL_SUMMON_AGATHA_FENRIS                  = 83982
+};
 
-    RegisterCreatureAI(npc_silverpine_horde_hauler);
-    RegisterSpellScript(spell_silverpine_magical_chains_hauler);
+// 83979 - Bond of the Val'kyr
+class spell_silverpine_bond_of_the_valkyr : public AuraScript
+{
+    PrepareAuraScript(spell_silverpine_bond_of_the_valkyr);
 
-    /* Forsaken High Command */
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_AGATHA_FENRIS });
+    }
 
-    RegisterCreatureAI(npc_silverpine_grand_executor_mortuus);
-    RegisterSpellScript(spell_silverpine_raise_forsaken_83173);
-    RegisterCreatureAI(npc_silverpine_fallen_human);
-    RegisterSpellScript(spell_silverpine_forsaken_trooper_masterscript_high_command);
-    RegisterCreatureAI(npc_silverpine_sylvanas_windrunner_high_command);
-    RegisterCreatureAI(npc_silverpine_deathstalker);
-    RegisterCreatureAI(npc_silverpine_worgen_renegade);
-    RegisterSpellScript(spell_silverpine_flurry_of_claws);
-    RegisterCreatureAI(npc_silverpine_forsaken_trooper);
-    RegisterCreatureAI(npc_silverpine_bat_handler_maggotbreath);
-    RegisterCreatureAI(npc_silverpine_forsaken_bat);
+    void AfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_SUMMON_AGATHA_FENRIS, true);
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->RemoveAura(SPELL_SUMMON_AGATHA_FENRIS);
+    }
+
+    void Register() override
+    {
+        AfterEffectApply += AuraEffectApplyFn(spell_silverpine_bond_of_the_valkyr::AfterApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_silverpine_bond_of_the_valkyr::AfterRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
```

Comment:
OnRemove

Target can be null in AfterRemove

Reply:
Feedback applied.